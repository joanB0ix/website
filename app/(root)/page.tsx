import SocialsSection from "@/app/(root)/about/SocialsSection";
import WhoamiSection from "@/app/(root)/about/WhoamiSection";
import ExperienceSection from "@/app/(root)/experience/ExperienceSection";
import ProjectsSection from "@/app/(root)/projects/ProjectsSection";
import { EXPERIENCE_LIST } from "@/config/experienceList";
import { PROJECT_LIST } from "@/config/projectList";
import { SOCIAL_LIST } from "@/config/socialList";
import { WHOAMI } from "@/config/whoami";

export default function Home() {
  return (
    <main className="flex-grow">
      <WhoamiSection content={WHOAMI} />
      <ExperienceSection experienceList={EXPERIENCE_LIST} />
      <ProjectsSection projectList={PROJECT_LIST} />
      <SocialsSection socialList={SOCIAL_LIST} />
    </main>
  );
}
