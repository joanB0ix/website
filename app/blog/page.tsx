import { Metadata } from "next";
import OverviewSection from "./overview/OverviewSection";

export const metadata: Metadata = {
  title: "Joan Boix's Blog",
  description: "This is my blog, where I write about things that interest me.",
  openGraph: {
    title: "Joan Boix's Blog",
    description:
      "This is my blog, where I write about things that interest me.",
    url: "https://joanboix.dev/blog",
    type: "website",
  },
  twitter: {
    card: "summary_large_image",
    title: "Joan Boix's Blog",
    description:
      "This is my blog, where I write about things that interest me.",
  },
};

export default function Blog() {
  return (
    <main className="flex-grow">
      <OverviewSection />
    </main>
  );
}
