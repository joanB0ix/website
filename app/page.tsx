import SocialsSection from "@/app/about/SocialsSection";
import WhoamiSection from "@/app/about/WhoamiSection";
import ExperienceSection from "@/app/experience/ExperienceSection";
import ProjectsSection from "@/app/projects/ProjectsSection";
import Footer from "@/components/Footer";
import Navbar from "@/components/Navbar";
import { EXPERIENCE_LIST } from "@/config/experienceList";
import { PROJECT_LIST } from "@/config/projectList";
import { SOCIAL_LIST } from "@/config/socialList";
import { WHOAMI } from "@/config/whoami";

export default function Home() {
  return (
    <div className="h-screen flex flex-col font-fira-code w-2/5 min-w-96 m-auto ">
      <Navbar />
      <main className="flex-grow">
        <WhoamiSection content={WHOAMI} />
        <ExperienceSection experienceList={EXPERIENCE_LIST} />
        <ProjectsSection projectList={PROJECT_LIST} />
        <SocialsSection socialList={SOCIAL_LIST} />
      </main>
      <Footer />
    </div>
  );
}
