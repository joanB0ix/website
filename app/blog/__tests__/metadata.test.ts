import { metadata } from "../page";

describe("Blog page metadata snapshot", () => {
  it("matches the expected metadata snapshot", () => {
    expect(metadata).toMatchSnapshot();
  });
});
