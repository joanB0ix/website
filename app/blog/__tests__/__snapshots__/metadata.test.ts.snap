// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog page metadata snapshot matches the expected metadata snapshot 1`] = `
{
  "description": "This is my blog, where I write about things that interest me.",
  "openGraph": {
    "description": "This is my blog, where I write about things that interest me.",
    "title": "Joan Boix's Blog",
    "type": "website",
    "url": "https://joanboix.dev/blog",
  },
  "title": "Joan Boix's Blog",
  "twitter": {
    "card": "summary_large_image",
    "description": "This is my blog, where I write about things that interest me.",
    "title": "Joan Boix's Blog",
  },
}
`;
